src/pip/_internal/network/utils.py

```diff
@@ -95,3 +96,8 @@ def response_chunks(response, chunk_size=CONTENT_CHUNK_SIZE):
             if not chunk:
                 break
             yield chunk
+    except (NewConnectionError, ReadTimeoutError) as exc:
+        raise NetworkConnectionError(
+            "Failed to get address for host! Check your network connectivity "
```

Comment:
`Failed to get address for host!` seems weird for a `ReadTimeoutError`

It definitely makes sense for `NewConnectionError` in the context of https://github.com/pypa/pip/issues/5380#issuecomment-513512674 but I don't think we should conflate the two exceptions types.